File: Assignments/Assignment 22/Java-Code/program22_5.java
// Accept N numbers from user and return frequancy of user defined in it.

import java.util.*;

class Logic
{
    int Check(int Arr[], int iLength, int iValue)
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            if(Arr[iCnt] == iValue)
            {
                iCount++;
            }
        }

        return iCount;
    }

}

class program22_5
{
    public static void main(String A[])
    { 
        Scanner sobj = new Scanner(System.in);
        Logic lobj = new Logic();

        int iSize = 0, iCnt = 0, iNo = 0;
        int iRet = 0;
        int Arr[];

        System.out.println("Enter number of elements : ");
        iSize = sobj.nextInt();

        System.out.println("Enter the element whose frequancy you want : ");
        iNo = sobj.nextInt();

        Arr = new int[iSize];

        System.out.println("Enter " + iSize + " elements : ");

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            System.out.println("Enter element no " + iCnt + " : ");
            Arr[iCnt] = sobj.nextInt();
        }

        iRet = lobj.Check(Arr, iSize, iNo);

        System.out.println("Frequency of " + iNo + " is : " + iRet);

    }
}

File: Assignments/Assignment 22/C-Code/program22_5.c
// Accept N numbers from user and return frequancy of user defined in it.

#include<stdio.h>
#include<stdlib.h>

int CheckFrequancy(int Arr[], int iLength, int iValue)
{
    int iCnt = 0;
    int iCount = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        if(Arr[iCnt] == iValue)
        {
            iCount++;
        }
    }

    return iCount;
}

int main()
{
    int iSize = 0, iCnt = 0;
    int iNo = 0;
    int iRet = 0;
    int *p= NULL;

    printf("Enter number of elements : \n");
    scanf("%d", &iSize);

    printf("Enter the element whose frequancy you want : \n");
    scanf("%d", &iNo);

    p = (int *)malloc(iSize*sizeof(int));

    if(p == NULL)
    {
        printf("Unable to allocate memory");
        return -1;
    }

    printf("Enter %d elements :\n", iSize);

    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element no %d : \n", iCnt+1);
        scanf("%d", &p[iCnt]);
    }

    iRet = CheckFrequancy(p, iSize, iNo);

    printf("Frequency of %d is : %d", iNo, iRet);
    
    free(p);

    return 0;
}